File: app/src/main/java/com/ocram/qichwadic/domain/repository/SearchRepository.java
package com.ocram.qichwadic.domain.repository;

import com.ocram.qichwadic.domain.model.Definition;
import com.ocram.qichwadic.domain.model.SearchResult;

import java.util.List;

import io.reactivex.Flowable;

public interface SearchRepository {

    Flowable<List<SearchResult>> searchOnline(int fromQuechua, String target, String word, int searchType);

    Flowable<List<SearchResult>> searchOffline(int fromQuechua, String target, String word, int searchType);

    Flowable<List<Definition>> fetchMoreResultsOffline(int dictionaryId, int searchType, String word, int page);

    Flowable<List<Definition>> fetchMoreResultsOnline(int dictionaryId, int searchType, String word, int page);
}

File: app/src/main/java/com/ocram/qichwadic/features/search/domain/repository/SearchRepository.kt
package com.ocram.qichwadic.features.search.domain.repository

import com.ocram.qichwadic.core.domain.model.DefinitionModel
import com.ocram.qichwadic.core.domain.model.SearchResultModel
import com.ocram.qichwadic.core.domain.model.SearchParams

interface SearchRepository {

    suspend fun search(searchParams: SearchParams): List<SearchResultModel>

    suspend fun fetchMoreResults(
        dictionaryId: Int,
        searchType: Int,
        word: String,
        page: Int
    ): List<DefinitionModel>
}